go/database/mpt/io/archive.go

```diff
@@ -100,6 +100,9 @@ func ExportArchive(directory string, out io.Writer) error {
 
 	// Encode diff of each individual block.
 	for block := uint64(0); block <= maxBlock; block++ {
+		if IsContextDone(ctx) {
+			return archive.Close()
```

Comment:
I think an abort by the context should result in an error since the export was not completed successfully.
Something linke:
```
return errors.Join(
    ErrCanceledByUser,
    archive.Close(),
)
```

Reply:
We had a discussion earlier with @kjezek and we agreed that it actually is not an error state. Should we still bring back the error?

Reply:
I would argue that it is an error, since the requested operation -- the export of the archive -- is not fulfilled and the caller should discard whatever was written into the output writer.

Reply:
thinking that the program can be actually killed by system, not only the user, would suggest that it is better to return an error

Reply:
Error added.